Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -3815,8 +3817,11 @@ void RenderBox::computeAndSetBlockDirectionMargins(const RenderBlock& containing
 LayoutUnit RenderBox::constrainBlockMarginInAvailableSpaceOrTrim(const RenderBox& containingBlock, LayoutUnit availableSpace, MarginTrimType marginSide) const
 {
     ASSERT(marginSide == MarginTrimType::BlockStart || marginSide == MarginTrimType::BlockEnd);
-    if (containingBlock.shouldTrimChildMargin(marginSide, *this))
+    if (containingBlock.shouldTrimChildMargin(marginSide, *this)) {
+        if (isGridItem() && marginSide == MarginTrimType::BlockStart)
+            const_cast<RenderBox&>(*this).markMarginAsTrimmed(MarginTrimType::BlockStart);
```

Comment:
please add a FIXME along the lines of, "This is supposed to be set when the trimmed margin is being...."

Reply:
done!